### PR TITLE
[FEATURE] - Layouting and routing

### PR DESCRIPTION
**Pull request by Anuraj-TR**
**Desktop view**

image

**************

**Tablet view**

image

**************

**Mobile view**

image

**************

Installed sass,React-router-dom,react-icons,tailwind

Implimented layouting,Routing

### CODE REVIEW

File: src/main/container/Layout.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Navbar } from "./Navbar";
+import { Sidebar } from "./Sidebar";
+
+export const Layout = ({ children }) => {
+  return (
+    <>
+      <div className="flex h-screen overflow-hidden justify-start relative">
+        <div className="absolute bottom-0 left-0 w-full bg-slate-500 z-10 md:static md:w-1/5 lg:w-1/6">
```
Comment (anandureghu) [inline src/main/container/Layout.jsx]: move this into sidebar

File: src/main/container/Layout.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          <Sidebar />
+        </div>
+        <div className="w-full relative">
```
Comment (anandureghu) [inline src/main/container/Layout.jsx:9]: move this into navbar

File: src/main/container/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import { useState } from "react";
+
+export const Sidebar = () => {
+  const [isTransitionActive, setIsTransitionActive] = useState(false);
+
+  return (
+    <>
+      <div className="w-full">
+        <ul className="w-full flex items-center justify-between p-2 md:flex-col ">
+          <li className="w-1/5 md:w-full md:mb-4">
```
Comment (anandureghu) [inline src/main/container/Sidebar.jsx]: make as a reusable component

Comment (anandureghu) [reply src/main/container/Sidebar.jsx]: eg:

 const listItems = [{
icon: <IconName>,
label: 'Label Text',
action: () => {},
}
]

File: src/main/container/Layout.jsx (hunk 1 above)
Comment (Anuraj-TR) [reply src/main/container/Layout.jsx]: moved into sidebar

File: src/main/container/Layout.jsx (hunk 2 above)
Comment (Anuraj-TR) [reply src/main/container/Layout.jsx:9]: done

File: src/main/container/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                Home
+              </div>
+            </Link>
+          </li>
+          <li className="min-w-14 min-h-14 -translate-y-1/2 border bg-slate-400 rounded-full md:hidden">
+            <button
+              className="w-full h-full  text-blue-500 text-center lg:flex lg:items-center lg:justify-center"
+              onClick={() => setIsTransitionActive(!isTransitionActive)}
+            >
+              {!isTransitionActive ? (
```
Comment (anandureghu) [inline src/main/container/Sidebar.jsx]: remove this

File: src/main/container/Sidebar.jsx (hunk 3 above)
Comment (Anuraj-TR) [reply src/main/container/Sidebar.jsx]: **created list items and made it reusable**

image


************************

image

_**!need help with adding active state on menu items**_

File: src/main/container/Sidebar.jsx (hunk 4 above)
Comment (Anuraj-TR) [reply src/main/container/Sidebar.jsx]: removed

File: src/main/container/Layout.jsx (hunk 1 above)
Comment (Anuraj-TR) [reply src/main/container/Layout.jsx]: and added new styles to sidebar

**Desktop**

image

**********************

**Tablet**

image


**********************

**Mobile**

image

**Review by anandureghu — APPROVED** [review]

**Pull request merged**